docs: Link to compatibility matrix for imageEnvoy (#7387)
hashicorp/consul: website/source/docs/platform/k8s/helm.html.md

  * <a name="v-connectinject-imageConsul" href="#v-connectinject-imageConsul">`imageConsul`</a> (`string: global.image`) - The name of the Docker image (including any tag) for Consul. This is used for proxy service registration, Envoy configuration, etc.

>   * <a name="v-connectinject-imageEnvoy" href="#v-connectinject-imageEnvoy">`imageEnvoy`</a> (`string: ""`) - The name of the Docker image (including any tag) for the Envoy sidecar. `envoy` must be on the executable path within this image. This Envoy version must be compatible with the Consul version used by the injector. This defaults to letting the injector choose the Envoy image, which is usually `envoyproxy/envoy-alpine`. Check [supported Envoy versions](https://www.consul.io/docs/connect/proxies/envoy.html#supported-versions) to ensure the version you are using is compatible with Consul.

Reviewer: ```suggestion
  * <a name="v-connectinject-imageEnvoy" href="#v-connectinject-imageEnvoy">`imageEnvoy`</a> (`string: ""`) - The name of the Docker image (including any tag) for the Envoy sidecar. `envoy` must be on the executable path within this image. This Envoy version must be compatible with the Consul version used by the injector. This defaults to letting the injector choose the Envoy image, which is usually `envoyproxy/envoy-alpine`. Check [supported Envoy versions](https://www.consul.io/docs/connect/proxies/envoy.html#supported-versions) to ensure the version you are using is compatible with your Consul version.
```

It looks like most other links to Consul docs on this page only use the relative path rather than the full url.

Reviewer: We've also added the complication of needing to check the `consul-k8s` version being used too. Somewhat related: [consul-helm #365](https://github.com/hashicorp/consul-helm/pull/365).

Reviewer 2: > This defaults to letting the injector choose the Envoy image, which is usually `envoyproxy/envoy-alpine`

It's actually never `envoyproxy/envoy-alpine` because that doesn't have a tag which means it's the same as `envoyproxy/envoy-alpine:latest`. If you want to put something here the best you could say would be
"For consul-k8s 0.12.0 the default is envoyproxy/envoy-alpine:v1.13.0.". But maybe it's best to just leave it as "the injector will use its default image".

Author: ```suggestion
  * <a name="v-connectinject-imageEnvoy" href="#v-connectinject-imageEnvoy">`imageEnvoy`</a> (`string: ""`) - The name of the Docker image (including any tag) for the Envoy sidecar. `envoy` must be on the executable path within this image. This Envoy version must be compatible with the Consul version used by the injector. If not specified this defaults to letting the injector choose the Envoy image. Check [supported Envoy versions](/docs/connect/proxies/envoy.html#supported-versions) to ensure the version you are using is compatible with Consul.
```

Author: @lkysow  Something like this?

Reviewer 2: Yeah that looks good!
